[Bug] [connectors-kafka] Base on SeaTunnel Engine Local Mode, kafka source is not working well

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

Base on SeaTunnel Engine Local Mode,I was able to run the following example with idea, then compiled and packaged it and ran it with an error.


### SeaTunnel Version

dev

### SeaTunnel Config

```conf
env {
  execution.parallelism = 1
  job.mode = "STREAMING"
  execution.checkpoint.interval = 5000
}

source {
  Kafka {
    result_table_name = "kafka"
    schema = {
      fields {
        name = "string"
        age = "int"
      }
    }
    format = text
    field_delimiter = "#"
    topic = "topic1"
    bootstrap.servers = "localhost:9092,"
    kafka.max.poll.records = 500
#     kafka.client.id = client_1
  }
}

transform {
    Split{
        source_table_name = "kafka"
        result_table_name = "split_res"
        separator = ","
        split_field = "name"
        output_fields = ["name1","name2"]
    }
}

sink {
  console {
    source_table_name="split_res"
  }
}
```


### Running Command

```shell
./bin/seatunnel.sh --config ./config/kafka2console.conf -e local
```


### Error Exception

```log
java.lang.ClassCastException: cannot assign instance of org.apache.kafka.common.TopicPartition to field org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaSourceSplit.topicPartition of type org.apache.kafka.common.TopicPartition in instance of org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaSourceSplit
        at java.io.ObjectStreamClass$FieldReflector.setObjFieldValues(ObjectStreamClass.java:2301) ~[?:1.8.0_352]
        at java.io.ObjectStreamClass.setObjFieldValues(ObjectStreamClass.java:1431) ~[?:1.8.0_352]
        at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2436) ~[?:1.8.0_352]
        at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2354) ~[?:1.8.0_352]
        at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2212) ~[?:1.8.0_352]
        at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1668) ~[?:1.8.0_352]
        at java.io.ObjectInputStream.readArray(ObjectInputStream.java:2118) ~[?:1.8.0_352]
        at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1656) ~[?:1.8.0_352]
        at java.io.ObjectInputStream.readObject(ObjectInputStream.java:502) ~[?:1.8.0_352]
        at java.io.ObjectInputStream.readObject(ObjectInputStream.java:460) ~[?:1.8.0_352]
        at org.apache.seatunnel.common.utils.SerializationUtils.deserialize(SerializationUtils.java:91) ~[seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at org.apache.seatunnel.engine.server.task.operation.source.AssignSplitOperation.lambda$run$1(AssignSplitOperation.java:57) ~[seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at org.apache.seatunnel.common.utils.RetryUtils.retryWithException(RetryUtils.java:47) ~[seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at org.apache.seatunnel.engine.server.task.operation.source.AssignSplitOperation.run(AssignSplitOperation.java:54) ~[seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:189) ~[seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:273) ~[seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:248) ~[seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:213) ~[seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:411) ~[seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:438) ~[seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:601) ~[seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:580) ~[seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:541) ~[seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:241) ~[seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:61) ~[seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at org.apache.seatunnel.engine.server.utils.NodeEngineUtil.sendOperationToMemberNode(NodeEngineUtil.java:40) ~[seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at org.apache.seatunnel.engine.server.execution.TaskExecutionContext.sendToMember(TaskExecutionContext.java:43) ~[seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at org.apache.seatunnel.engine.server.task.context.SeaTunnelSplitEnumeratorContext.assignSplit(SeaTunnelSplitEnumeratorContext.java:55) ~[seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at java.util.HashMap.forEach(HashMap.java:1290) [?:1.8.0_352]
        at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaSourceSplitEnumerator.assignSplit(KafkaSourceSplitEnumerator.java:251) [connector-kafka-2.3.1-SNAPSHOT.jar:2.3.1-SNAPSHOT]
        at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaSourceSplitEnumerator.run(KafkaSourceSplitEnumerator.java:115) [connector-kafka-2.3.1-SNAPSHOT.jar:2.3.1-SNAPSHOT]
        at org.apache.seatunnel.engine.server.task.SourceSplitEnumeratorTask.stateProcess(SourceSplitEnumeratorTask.java:238) [seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at org.apache.seatunnel.engine.server.task.SourceSplitEnumeratorTask.call(SourceSplitEnumeratorTask.java:123) [seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at org.apache.seatunnel.engine.server.TaskExecutionService$BlockingWorker.run(TaskExecutionService.java:350) [seatunnel-starter.jar:2.3.1-SNAPSHOT]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_352]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_352]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_352]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_352]
        at java.lang.Thread.run(Thread.java:750) [?:1.8.0_352]
```


### Flink or Spark Version

_No response_

### Java or Scala Version

Java 1.8

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

> 这些是本地模式的一些错误，我将修复这个问题。你可以尝试使用集群模式。它工作得很好。



> These some bug with local mode, I will fix this. No changes to the configuration file. You can try use cluster mode. It work fine.

When I use cluster mode, I get the following error. Can you help me with this?
`
2022-12-26 14:39:42,699 INFO  org.apache.seatunnel.engine.server.TaskExecutionService - [localhost]:5801 [seatunnel_default_cluster] [5.1] Task TaskGroupLocation{jobId=659646490632781825, pipelineId=1, taskGroupId=30000} complete with state FAILED
2022-12-26 14:39:42,705 INFO  org.apache.seatunnel.engine.server.dag.physical.PhysicalVertex - Job SeaTunnel (659646490632781825), Pipeline: [(1/1)], task: [Kafka-SourceTask (1/1)] turn to end state FAILED.
2022-12-26 14:39:42,705 ERROR org.apache.seatunnel.engine.server.dag.physical.PhysicalVertex - Job SeaTunnel (659646490632781825), Pipeline: [(1/1)], task: [Kafka-SourceTask (1/1)] end with state FAILED and Exception: java.util.concurrent.CompletionException: java.lang.RuntimeException: Thread %s has no metrics context set, this method can be called only on threads executing the job's processors
        at java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:375)
        at java.util.concurrent.CompletableFuture.join(CompletableFuture.java:1947)
        at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaSourceReader.lambda$pollNext$4(KafkaSourceReader.java:151)
        at java.lang.Iterable.forEach(Iterable.java:75)
        at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaSourceReader.pollNext(KafkaSourceReader.java:109)
        at org.apache.seatunnel.engine.server.task.flow.SourceFlowLifeCycle.collect(SourceFlowLifeCycle.java:115)
        at org.apache.seatunnel.engine.server.task.SourceSeaTunnelTask.collect(SourceSeaTunnelTask.java:68)
        at org.apache.seatunnel.engine.server.task.SeaTunnelTask.stateProcess(SeaTunnelTask.java:157)
        at org.apache.seatunnel.engine.server.task.SourceSeaTunnelTask.call(SourceSeaTunnelTask.java:74)
        at org.apache.seatunnel.engine.server.TaskExecutionService$BlockingWorker.run(TaskExecutionService.java:350)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:750)
Caused by: java.lang.RuntimeException: Thread %s has no metrics context set, this method can be called only on threads executing the job's processors
        at org.apache.seatunnel.engine.server.metrics.MetricsImpl.getContext(MetricsImpl.java:50)
        at org.apache.seatunnel.engine.server.metrics.MetricsImpl.qpsMetric(MetricsImpl.java:39)
        at org.apache.seatunnel.engine.server.metrics.Metrics.qpsMetric(Metrics.java:41)
        at org.apache.seatunnel.engine.server.task.SeaTunnelSourceCollector.collect(SeaTunnelSourceCollector.java:47)
        at org.apache.seatunnel.api.serialization.DeserializationSchema.deserialize(DeserializationSchema.java:39)
        at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaSourceReader.lambda$null$3(KafkaSourceReader.java:126)
        at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaConsumerThread.run(KafkaConsumerThread.java:53)
        ... 5 more
`

> > These some bug with local mode, I will fix this. You can try use cluster mode. It work fine.
> 
> ok, I will try it.

When I use cluster mode, I get the following error. Can you help me with this?

```
2022-12-26 14:39:42,699 INFO  org.apache.seatunnel.engine.server.TaskExecutionService - [localhost]:5801 [seatunnel_default_cluster] [5.1] Task TaskGroupLocation{jobId=659646490632781825, pipelineId=1, taskGroupId=30000} complete with state FAILED
2022-12-26 14:39:42,705 INFO  org.apache.seatunnel.engine.server.dag.physical.PhysicalVertex - Job SeaTunnel (659646490632781825), Pipeline: [(1/1)], task: [Kafka-SourceTask (1/1)] turn to end state FAILED.
2022-12-26 14:39:42,705 ERROR org.apache.seatunnel.engine.server.dag.physical.PhysicalVertex - Job SeaTunnel (659646490632781825), Pipeline: [(1/1)], task: [Kafka-SourceTask (1/1)] end with state FAILED and Exception: java.util.concurrent.CompletionException: java.lang.RuntimeException: Thread %s has no metrics context set, this method can be called only on threads executing the job's processors
        at java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:375)
        at java.util.concurrent.CompletableFuture.join(CompletableFuture.java:1947)
        at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaSourceReader.lambda$pollNext$4(KafkaSourceReader.java:151)
        at java.lang.Iterable.forEach(Iterable.java:75)
        at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaSourceReader.pollNext(KafkaSourceReader.java:109)
        at org.apache.seatunnel.engine.server.task.flow.SourceFlowLifeCycle.collect(SourceFlowLifeCycle.java:115)
        at org.apache.seatunnel.engine.server.task.SourceSeaTunnelTask.collect(SourceSeaTunnelTask.java:68)
        at org.apache.seatunnel.engine.server.task.SeaTunnelTask.stateProcess(SeaTunnelTask.java:157)
        at org.apache.seatunnel.engine.server.task.SourceSeaTunnelTask.call(SourceSeaTunnelTask.java:74)
        at org.apache.seatunnel.engine.server.TaskExecutionService$BlockingWorker.run(TaskExecutionService.java:350)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:750)
Caused by: java.lang.RuntimeException: Thread %s has no metrics context set, this method can be called only on threads executing the job's processors
        at org.apache.seatunnel.engine.server.metrics.MetricsImpl.getContext(MetricsImpl.java:50)
        at org.apache.seatunnel.engine.server.metrics.MetricsImpl.qpsMetric(MetricsImpl.java:39)
        at org.apache.seatunnel.engine.server.metrics.Metrics.qpsMetric(Metrics.java:41)
        at org.apache.seatunnel.engine.server.task.SeaTunnelSourceCollector.collect(SeaTunnelSourceCollector.java:47)
        at org.apache.seatunnel.api.serialization.DeserializationSchema.deserialize(DeserializationSchema.java:39)
        at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaSourceReader.lambda$null$3(KafkaSourceReader.java:126)
        at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaConsumerThread.run(KafkaConsumerThread.java:53)
        ... 5 more
```

> These some bug with local mode, I will fix this. You can try use cluster mode. It work fine.

When I use cluster mode, I get the following error. Can you help me with this?
```
2022-12-26 14:39:42,699 INFO  org.apache.seatunnel.engine.server.TaskExecutionService - [localhost]:5801 [seatunnel_default_cluster] [5.1] Task TaskGroupLocation{jobId=659646490632781825, pipelineId=1, taskGroupId=30000} complete with state FAILED
2022-12-26 14:39:42,705 INFO  org.apache.seatunnel.engine.server.dag.physical.PhysicalVertex - Job SeaTunnel (659646490632781825), Pipeline: [(1/1)], task: [Kafka-SourceTask (1/1)] turn to end state FAILED.
2022-12-26 14:39:42,705 ERROR org.apache.seatunnel.engine.server.dag.physical.PhysicalVertex - Job SeaTunnel (659646490632781825), Pipeline: [(1/1)], task: [Kafka-SourceTask (1/1)] end with state FAILED and Exception: java.util.concurrent.CompletionException: java.lang.RuntimeException: Thread %s has no metrics context set, this method can be called only on threads executing the job's processors
        at java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:375)
        at java.util.concurrent.CompletableFuture.join(CompletableFuture.java:1947)
        at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaSourceReader.lambda$pollNext$4(KafkaSourceReader.java:151)
        at java.lang.Iterable.forEach(Iterable.java:75)
        at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaSourceReader.pollNext(KafkaSourceReader.java:109)
        at org.apache.seatunnel.engine.server.task.flow.SourceFlowLifeCycle.collect(SourceFlowLifeCycle.java:115)
        at org.apache.seatunnel.engine.server.task.SourceSeaTunnelTask.collect(SourceSeaTunnelTask.java:68)
        at org.apache.seatunnel.engine.server.task.SeaTunnelTask.stateProcess(SeaTunnelTask.java:157)
        at org.apache.seatunnel.engine.server.task.SourceSeaTunnelTask.call(SourceSeaTunnelTask.java:74)
        at org.apache.seatunnel.engine.server.TaskExecutionService$BlockingWorker.run(TaskExecutionService.java:350)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:750)
Caused by: java.lang.RuntimeException: Thread %s has no metrics context set, this method can be called only on threads executing the job's processors
        at org.apache.seatunnel.engine.server.metrics.MetricsImpl.getContext(MetricsImpl.java:50)
        at org.apache.seatunnel.engine.server.metrics.MetricsImpl.qpsMetric(MetricsImpl.java:39)
        at org.apache.seatunnel.engine.server.metrics.Metrics.qpsMetric(Metrics.java:41)
        at org.apache.seatunnel.engine.server.task.SeaTunnelSourceCollector.collect(SeaTunnelSourceCollector.java:47)
        at org.apache.seatunnel.api.serialization.DeserializationSchema.deserialize(DeserializationSchema.java:39)
        at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaSourceReader.lambda$null$3(KafkaSourceReader.java:126)
        at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaConsumerThread.run(KafkaConsumerThread.java:53)
        ... 5 more
```

> > These some bug with local mode, I will fix this. You can try use cluster mode. It work fine.
> 
> When I use cluster mode, I get the following error. Can you help me with this?
> 
> ```
> 2022-12-26 14:39:42,699 INFO  org.apache.seatunnel.engine.server.TaskExecutionService - [localhost]:5801 [seatunnel_default_cluster] [5.1] Task TaskGroupLocation{jobId=659646490632781825, pipelineId=1, taskGroupId=30000} complete with state FAILED
> 2022-12-26 14:39:42,705 INFO  org.apache.seatunnel.engine.server.dag.physical.PhysicalVertex - Job SeaTunnel (659646490632781825), Pipeline: [(1/1)], task: [Kafka-SourceTask (1/1)] turn to end state FAILED.
> 2022-12-26 14:39:42,705 ERROR org.apache.seatunnel.engine.server.dag.physical.PhysicalVertex - Job SeaTunnel (659646490632781825), Pipeline: [(1/1)], task: [Kafka-SourceTask (1/1)] end with state FAILED and Exception: java.util.concurrent.CompletionException: java.lang.RuntimeException: Thread %s has no metrics context set, this method can be called only on threads executing the job's processors
>         at java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:375)
>         at java.util.concurrent.CompletableFuture.join(CompletableFuture.java:1947)
>         at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaSourceReader.lambda$pollNext$4(KafkaSourceReader.java:151)
>         at java.lang.Iterable.forEach(Iterable.java:75)
>         at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaSourceReader.pollNext(KafkaSourceReader.java:109)
>         at org.apache.seatunnel.engine.server.task.flow.SourceFlowLifeCycle.collect(SourceFlowLifeCycle.java:115)
>         at org.apache.seatunnel.engine.server.task.SourceSeaTunnelTask.collect(SourceSeaTunnelTask.java:68)
>         at org.apache.seatunnel.engine.server.task.SeaTunnelTask.stateProcess(SeaTunnelTask.java:157)
>         at org.apache.seatunnel.engine.server.task.SourceSeaTunnelTask.call(SourceSeaTunnelTask.java:74)
>         at org.apache.seatunnel.engine.server.TaskExecutionService$BlockingWorker.run(TaskExecutionService.java:350)
>         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
>         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
>         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
>         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
>         at java.lang.Thread.run(Thread.java:750)
> Caused by: java.lang.RuntimeException: Thread %s has no metrics context set, this method can be called only on threads executing the job's processors
>         at org.apache.seatunnel.engine.server.metrics.MetricsImpl.getContext(MetricsImpl.java:50)
>         at org.apache.seatunnel.engine.server.metrics.MetricsImpl.qpsMetric(MetricsImpl.java:39)
>         at org.apache.seatunnel.engine.server.metrics.Metrics.qpsMetric(Metrics.java:41)
>         at org.apache.seatunnel.engine.server.task.SeaTunnelSourceCollector.collect(SeaTunnelSourceCollector.java:47)
>         at org.apache.seatunnel.api.serialization.DeserializationSchema.deserialize(DeserializationSchema.java:39)
>         at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaSourceReader.lambda$null$3(KafkaSourceReader.java:126)
>         at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaConsumerThread.run(KafkaConsumerThread.java:53)
>         ... 5 more
> ```

@chaorongzhi  Are you testing the latest dev branch, if not update the code. This error was solved before

> > > These some bug with local mode, I will fix this. You can try use cluster mode. It work fine.
> > 
> > 
> > When I use cluster mode, I get the following error. Can you help me with this?
> > ```
> > 2022-12-26 14:39:42,699 INFO  org.apache.seatunnel.engine.server.TaskExecutionService - [localhost]:5801 [seatunnel_default_cluster] [5.1] Task TaskGroupLocation{jobId=659646490632781825, pipelineId=1, taskGroupId=30000} complete with state FAILED
> > 2022-12-26 14:39:42,705 INFO  org.apache.seatunnel.engine.server.dag.physical.PhysicalVertex - Job SeaTunnel (659646490632781825), Pipeline: [(1/1)], task: [Kafka-SourceTask (1/1)] turn to end state FAILED.
> > 2022-12-26 14:39:42,705 ERROR org.apache.seatunnel.engine.server.dag.physical.PhysicalVertex - Job SeaTunnel (659646490632781825), Pipeline: [(1/1)], task: [Kafka-SourceTask (1/1)] end with state FAILED and Exception: java.util.concurrent.CompletionException: java.lang.RuntimeException: Thread %s has no metrics context set, this method can be called only on threads executing the job's processors
> >         at java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:375)
> >         at java.util.concurrent.CompletableFuture.join(CompletableFuture.java:1947)
> >         at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaSourceReader.lambda$pollNext$4(KafkaSourceReader.java:151)
> >         at java.lang.Iterable.forEach(Iterable.java:75)
> >         at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaSourceReader.pollNext(KafkaSourceReader.java:109)
> >         at org.apache.seatunnel.engine.server.task.flow.SourceFlowLifeCycle.collect(SourceFlowLifeCycle.java:115)
> >         at org.apache.seatunnel.engine.server.task.SourceSeaTunnelTask.collect(SourceSeaTunnelTask.java:68)
> >         at org.apache.seatunnel.engine.server.task.SeaTunnelTask.stateProcess(SeaTunnelTask.java:157)
> >         at org.apache.seatunnel.engine.server.task.SourceSeaTunnelTask.call(SourceSeaTunnelTask.java:74)
> >         at org.apache.seatunnel.engine.server.TaskExecutionService$BlockingWorker.run(TaskExecutionService.java:350)
> >         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
> >         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
> >         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> >         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> >         at java.lang.Thread.run(Thread.java:750)
> > Caused by: java.lang.RuntimeException: Thread %s has no metrics context set, this method can be called only on threads executing the job's processors
> >         at org.apache.seatunnel.engine.server.metrics.MetricsImpl.getContext(MetricsImpl.java:50)
> >         at org.apache.seatunnel.engine.server.metrics.MetricsImpl.qpsMetric(MetricsImpl.java:39)
> >         at org.apache.seatunnel.engine.server.metrics.Metrics.qpsMetric(Metrics.java:41)
> >         at org.apache.seatunnel.engine.server.task.SeaTunnelSourceCollector.collect(SeaTunnelSourceCollector.java:47)
> >         at org.apache.seatunnel.api.serialization.DeserializationSchema.deserialize(DeserializationSchema.java:39)
> >         at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaSourceReader.lambda$null$3(KafkaSourceReader.java:126)
> >         at org.apache.seatunnel.connectors.seatunnel.kafka.source.KafkaConsumerThread.run(KafkaConsumerThread.java:53)
> >         ... 5 more
> > ```
> 
> @chaorongzhi Are you testing the latest dev branch, if not update the code. This error was solved before

@ic4y I tested it four days ago, I'll test it again soon. thank you.

--- a/seatunnel-engine/seatunnel-engine-core/src/main/java/org/apache/seatunnel/engine/core/parse/JobConfigParser.java
+++ b/seatunnel-engine/seatunnel-engine-core/src/main/java/org/apache/seatunnel/engine/core/parse/JobConfigParser.java
@@ -34,6 +34,7 @@ import org.apache.seatunnel.common.constants.JobMode;
 import org.apache.seatunnel.core.starter.config.ConfigBuilder;
 import org.apache.seatunnel.engine.common.config.JobConfig;
 import org.apache.seatunnel.engine.common.exception.JobDefineCheckException;
+import org.apache.seatunnel.engine.common.loader.SeatunnelChildFirstClassLoader;
 import org.apache.seatunnel.engine.common.utils.IdGenerator;
 import org.apache.seatunnel.engine.core.dag.actions.Action;
 import org.apache.seatunnel.engine.core.dag.actions.SinkAction;
@@ -108,6 +109,7 @@ public class JobConfigParser {
     }
 
     public ImmutablePair<List<Action>, Set<URL>> parse() {
+        Thread.currentThread().setContextClassLoader(new SeatunnelChildFirstClassLoader(new ArrayList<>()));
         List<? extends Config> sinkConfigs = seaTunnelJobConfig.getConfigList("sink");
         List<? extends Config> transformConfigs =
             TypesafeConfigUtils.getConfigList(seaTunnelJobConfig, "transform", Collections.emptyList());

--- a/seatunnel-plugin-discovery/src/main/java/org/apache/seatunnel/plugin/discovery/AbstractPluginDiscovery.java
+++ b/seatunnel-plugin-discovery/src/main/java/org/apache/seatunnel/plugin/discovery/AbstractPluginDiscovery.java
@@ -71,7 +71,7 @@ public abstract class AbstractPluginDiscovery<T> implements PluginDiscovery<T> {
      * Add jar url to classloader. The different engine should have different logic to add url into
      * their own classloader
      */
-    private static final BiConsumer DEFAULT_URL_TO_CLASSLOADER = (classLoader, url) -> {
+    private static final BiConsumer<ClassLoader, URL> DEFAULT_URL_TO_CLASSLOADER = (classLoader, url) -> {
         if (classLoader instanceof URLClassLoader) {
             ReflectionUtils.invoke(classLoader, "addURL", url);
         } else {
@@ -210,7 +210,7 @@ public abstract class AbstractPluginDiscovery<T> implements PluginDiscovery<T> {
         if (this.pluginDir.toFile().exists()) {
             log.info("load plugin from plugin dir: {}", this.pluginDir);
             List<URL> files = FileUtils.searchJarFiles(this.pluginDir);
-            factories = FactoryUtil.discoverFactories(new URLClassLoader((URL[]) files.toArray(new URL[0])));
+            factories = FactoryUtil.discoverFactories(new URLClassLoader(files.toArray(new URL[0])));
         } else {
             log.info("plugin dir: {} not exists, load plugin from classpath", this.pluginDir);
             factories = FactoryUtil.discoverFactories(Thread.currentThread().getContextClassLoader());
@@ -220,9 +220,7 @@ public abstract class AbstractPluginDiscovery<T> implements PluginDiscovery<T> {
 
         factories.forEach(plugin -> {
             if (TableSourceFactory.class.isAssignableFrom(plugin.getClass())) {
-                if (plugins.get(PluginType.SOURCE) == null) {
-                    plugins.put(PluginType.SOURCE, new LinkedHashMap());
-                }
+                plugins.computeIfAbsent(PluginType.SOURCE, k -> new LinkedHashMap<>());
 
                 plugins.get(PluginType.SOURCE).put(PluginIdentifier.of(
                         "seatunnel",
@@ -234,9 +232,7 @@ public abstract class AbstractPluginDiscovery<T> implements PluginDiscovery<T> {
             }
 
             if (TableSinkFactory.class.isAssignableFrom(plugin.getClass())) {
-                if (plugins.get(PluginType.SINK) == null) {
-                    plugins.put(PluginType.SINK, new LinkedHashMap());
-                }
+                plugins.computeIfAbsent(PluginType.SINK, k -> new LinkedHashMap<>());
 
                 plugins.get(PluginType.SINK).put(PluginIdentifier.of(
                         "seatunnel",
@@ -248,9 +244,7 @@ public abstract class AbstractPluginDiscovery<T> implements PluginDiscovery<T> {
             }
 
             if (TableTransformFactory.class.isAssignableFrom(plugin.getClass())) {
-                if (plugins.get(PluginType.TRANSFORM) == null) {
-                    plugins.put(PluginType.TRANSFORM, new LinkedHashMap());
-                }
+                plugins.computeIfAbsent(PluginType.TRANSFORM, k -> new LinkedHashMap<>());
 
                 plugins.get(PluginType.TRANSFORM).put(PluginIdentifier.of(
                         "seatunnel",

